### DISCUSSION TITLE
What is the largest value supported by this library given in Unix time?

### DISCUSSION THREAD

**Discussion by kbakdev** [js-fns/date-fns#3407] [General] [answered]

**Comment by fturmel** [comment, marked as answer]
Everything is built on top of the native `Date`, so you can look up the specs for details. Timestamps are milliseconds since epoch.

> A Number can exactly represent all [integers](https://tc39.es/ecma262/multipage/notational-conventions.html#integer) from -9,007,199,254,740,992 to 9,007,199,254,740,992 ([21.1.2.8](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-number.min_safe_integer) and [21.1.2.6](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-number.max_safe_integer)). A time value supports a slightly smaller range of -8,640,000,000,000,000 to 8,640,000,000,000,000 milliseconds. This yields a supported time value range of exactly -100,000,000 days to 100,000,000 days relative to midnight at the beginning of 1 January 1970 UTC.

https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-time-values-and-time-range
